Jellyfin.Api/Helpers/AudioHelper.cs

```diff
@@ -154,4 +159,44 @@ public async Task<ActionResult> GetAudioStream(
             transcodingJobType,
             cancellationTokenSource).ConfigureAwait(false);
     }
+
+    /// <summary>
+    /// Gets audio data to draw a waveform bar.
+    /// </summary>
+    /// <param name="itemId">The item id.</param>
+    /// <returns>A <see cref="FileStream"/> containing the resulting .dat file.</returns>
+    public async Task<FileStream> GetAudioWaveForm(Guid itemId)
+    {
+        var item = _libraryManager.GetItemById<BaseItem>(itemId)
+            ?? throw new ResourceNotFoundException();
+
+        var libraryOptions = BaseItem.LibraryManager.GetLibraryOptions(item);
+        var saveInMediaFolder = libraryOptions.SaveLocalMetadata;
+        var saveFileName = Path.GetFileNameWithoutExtension(item.Path) + ".csv";
+        string outputPath = saveInMediaFolder
+            ? Path.Combine(item.ContainingFolderPath, saveFileName)
+            : Path.Combine(item.GetInternalMetadataPath(), saveFileName);
+
+        if (!File.Exists(outputPath))
+        {
+            var processStartInfo = new ProcessStartInfo("ffprobe", $"-v error -select_streams a:0 -show_entries stream=sample_rate -of csv=p=0 \"{item.Path}\"") { RedirectStandardOutput = true, UseShellExecute = false };
+            int sampleRate = 0;
+            using (var process = Process.Start(processStartInfo))
+            {
+                if (process != null)
+                {
+                    string output = await process.StandardOutput.ReadToEndAsync().ConfigureAwait(false);
+                    // This value sets the length of a frame, the amount of measurements per second.
+                    int samplesPerSecond = 2;
+                    sampleRate = Math.Max(int.Parse(output, System.Globalization.CultureInfo.InvariantCulture) / samplesPerSecond, 1);
+                }
+            }
+
+            await Task.Run(() => Process.Start("ffprobe", $"-v error -f lavfi -i \"amovie={item.Path},asetnsamples={sampleRate},astats=metadata=1:reset=1\" -show_entries frame_tags=lavfi.astats.Overall.RMS_peak -of csv=p=0 -o \"{outputPath}\"").WaitForExit()).ConfigureAwait(false);
```

Comment:
The path should be escaped carefully because all kind of special characters that will crash the cli may exist